engine/flink/components/table/src/main/scala/pl/touk/nussknacker/engine/flink/table/source/FlinkMiniClusterTableOperations.scala

```diff
@@ -22,36 +24,36 @@ import scala.util.{Failure, Success, Try, Using}
 object FlinkMiniClusterTableOperations extends LazyLogging {
 
   // TODO: check the minicluster releases memory properly and if not, refactor to reuse one minicluster per all usages
-  def generateTestData(amountOfRecordsToGenerate: Int, schema: Schema): TestData = {
+  def generateTestData(amountOfRecordsToGenerate: Int, schema: Schema, mode: DataGenerationMode): TestData =
     // setting context classloader because Flink in multiple places relies on it and without this temporary override it doesnt have
     // the necessary classes
     ThreadUtils.withThisAsContextClassLoader(getClass.getClassLoader) {
-      implicit val env: TableEnvironment    = MiniClusterEnvBuilder.buildTableEnv
-      val (inputTableName, outputTableName) = generateTestDataTableNames
-      createRandomDataGeneratorTable(amountOfRecordsToGenerate, schema, inputTableName)
-      val outputFilePath = createTempFileTable(schema, outputTableName)
+      implicit val env: TableEnvironment = MiniClusterEnvBuilder.buildTableEnv
+      val inputTableName = mode match {
+        case RandomDataMode                         => createRandomDataGeneratorTable(amountOfRecordsToGenerate, schema)
+        case RealDataMode(sqlStatements, tableName) => createRealDataGeneratorTable(sqlStatements, tableName)
+      }
+      val (outputFilePath, outputTableName) = createTempFileTable(schema)
       val generatedRows = Try {
         insertDataAndAwait(inputTableName, outputTableName)
         readRecordsFromFilesUnderPath(outputFilePath)
       }
-      cleanup(outputFilePath, List(inputTableName, outputTableName))
+      cleanup(outputFilePath)
       val rows = generatedRows.get
       TestData(rows.map(TestRecord(_)))
     }
-  }
 
   def parseTestRecords(records: List[TestRecord], schema: Schema): List[RECORD] =
     ThreadUtils.withThisAsContextClassLoader(getClass.getClassLoader) {
       implicit val env: StreamTableEnvironment = MiniClusterEnvBuilder.buildStreamTableEnv
-      val (inputTableName, _)                  = generateTestDataTableNames
-      val inputTablePath                       = createTempFileTable(schema, inputTableName)
+      val (inputTablePath, inputTableName)     = createTempFileTable(schema)
```

Comment:
This: `env.from(inputTableName)` is wrong. You should escape the `inputTableName`. It would be nice to reuse `Table` object returned in `executeSqlAndGetTable`.